### PR TITLE
Add metrics for migration status

### PR DESCRIPTION
**Pull request by caesarxuchao**
Sample metrics output: https://pastebin.com/TPdHLvCd

/assign @yliaog @logicalhan

### CODE REVIEW

File: pkg/migrator/metrics/metrics.go [hunk 1]
```diff
@@ -32,20 +32,21 @@ var (
 
 // CoreMigratorMetrics instruments core migrator with prometheus metrics.
 type CoreMigratorMetrics struct {
-	migrated  *prometheus.CounterVec
-	remaining *prometheus.GaugeVec
+	objectsMigrated *prometheus.CounterVec
+	remaining       *prometheus.GaugeVec
```
Comment (yliaog) [inline pkg/migrator/metrics/metrics.go]: probably better to rename it to objectsRemaining or sth? previously both migrated and remaining are about objects, now 'migration' is added, better to tell remaining is about 'objects' in the name

Comment (caesarxuchao) [reply pkg/migrator/metrics/metrics.go]: Good point. Updated.

File: pkg/migrator/metrics/metrics.go [hunk 2]
```diff
@@ -32,47 +32,69 @@ var (
...
-	prometheus.MustRegister(remaining)
+	prometheus.MustRegister(objectsRemaining)
+
+	migration := prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Namespace: namespace,
+			Subsystem: subsystem,
+			Name:      "migrations",
+			Help:      "The number of completed migration, labeled with the full resource name, and the status of the migration (failed or succeeded)",
+		}, []string{"resource", "Status"})
```
Comment (logicalhan) [inline pkg/migrator/metrics/metrics.go]: Why is one of these labels capitalized?

**Pull request merged**